.github/actions/setup-linux/action.yml

```
@@ -26,8 +26,19 @@ runs:
         echo "instance-type: $(get_ec2_metadata instance-type)"
         echo "system info $(uname -a)"
 
+    - name: Check if in a ARC runner
+      shell: bash
+      id: check_arc_runner
+      run: |
+        if [[ -f /.inarc ]] ; then
+          echo "IN_ARC_RUNNER=true" >> $GITHUB_OUTPUT
+        else
+          echo "IN_ARC_RUNNER=false" >> $GITHUB_OUTPUT
+        fi
```

I'm not a shell expert, but there must be a more compact way of doing it, for example
```suggestion
        echo "IN_ARC_RUNNER=$(if [ -f /.inarc ]; then echo true; else echo false; fi)"  >> $GITHUB_OUTPUT
```

---

it is the trade off between cleverness and readability. But if the goal is being clever, I believe this is the best way to do it:

```
echo "IN_ARC_RUNNER=$([ -f /.inarc ] && echo true || echo false)"  >> $GITHUB_OUTPUT
```